### ISSUE TITLE
The Broccoli Plugin: [TemplateCompiler] failed with: undefined

### ISSUE THREAD

**Issue by aaroncraigongithub** [Vestorly/ember-cli-emblem#47]
My steps:

```
$ ember install ember-cli-emblem
```

Create a template `foo.emblem`

```
$ ember s
```

Did I miss a step?

**Comment by aaroncraigongithub** [comment]
Tried again and this is now working.  Not sure why but I'll close this for now.

**Comment by phortx** [comment]
I have these errors too at strange conditions.

```emblem
h4.card-title
  strong
    link-to "thingy.show" thingy.id class="white-text"
      = thingy.name
```

Doesn't work: `The Broccoli Plugin: [broccoli-persistent-filter:TemplateCompiler] failed with: undefined`

But that works: 

```emblem
h4.card-title: strong: link-to "thingy.show" thingy.id class="white-text"
  = thingy.name
```

Any idea?

**Comment by phortx** [comment]
And this code produces errors too:

```slim
h3
   | Some Text
   span class="pull-right mx-1 badge" some badge
```

**Comment by phortx** [comment]
And this one too:

`shared/modal-popup id=(concat "modal-" thingy.id) title=thingy.name class='modal-bottom-center' register-as=modal as |m|`

But without the `id`, it works.

Stack trace is:

```
file changed templates/components/shared/thingy-card.em
DEPRECATION: Explicit component declarations will be interpreted as angle-bracket components in a later release: i.icon.ion-checkmark
File: frontend/templates/components/shared/thingy-card.em
The Broccoli Plugin: [broccoli-persistent-filter:TemplateCompiler] failed with:
undefined

The broccoli plugin was instantiated at:
Error
    at TemplateCompiler.Plugin (node_modules/broccoli-plugin/index.js:7:31)
    at TemplateCompiler.Filter [as constructor] (node_modules/broccoli-persistent-filter/index.js:64:10)
    at TemplateCompiler (node_modules/ember-cli-htmlbars/index.js:32:5)
    at Object.toTree (node_modules/ember-cli-htmlbars/ember-addon-main.js:23:16)
    at node_modules/ember-cli-preprocess-registry/preprocessors.js:180:26
    at Array.forEach (<anonymous>)
    at processPlugins (node_modules/ember-cli-preprocess-registry/preprocessors.js:178:11)
    at module.exports.preprocessTemplates (node_modules/ember-cli-preprocess-registry/preprocessors.js:161:10)
    at EmberApp._processedTemplatesTree (node_modules/ember-cli/lib/broccoli/ember-app.js:891:50)
    at EmberApp.appAndDependencies (node_modules/ember-cli/lib/broccoli/ember-app.js:1109:26)
```

Could you reopen that issue?

**Comment by thec0keman** [comment]
That error message is from the pegjs compiler, which is what converts emblem syntax to handlebars.  Unfortunately, helpful errors are not implemented yet.  (At least, there are some tools within pegjs that would give better feedback on why the compilation failure happened.)

I can see you are on the latest emblem, and those all look like valid Emblem syntax.  They are working in the web preview:

[Example 1](https://mammothhr.github.io/try-emblem/#/?result=h4.card-title%0A%20%20strong%0A%20%20%20%20link-to%20%22thingy.show%22%20thingy.id%20class%3D%22white-text%22%0A%20%20%20%20%20%20%3D%20thingy.name)
[Example 2](https://mammothhr.github.io/try-emblem/#/?result=h3%0A%20%20%20%7C%20Some%20Text%0A%20%20%20span%20class%3D%22pull-right%20mx-1%20badge%22%20some%20badge)
[Example 3](https://mammothhr.github.io/try-emblem/#/?result=shared%2Fmodal-popup%20id%3D(concat%20%22modal-%22%20thingy.id)%20title%3Dthingy.name%20class%3D'modal-bottom-center'%20register-as%3Dmodal%20as%20%7Cm%7C)

I'd suggest clearing your `node_modules` and reinstalling and see if that helps.  With those template snippets were there any other contextual pieces?  also, are you using npm or yarn?

**Comment by phortx** [comment]
Thank you, for your answer @thec0keman !

I'm using yarn (not flat). Removing `node_modules` and running `yarn` did not help.

Is there any way to debug that? I've tried to use the debugging flag, but that didn't provide any helpful information.

I will try to change the code to reduce it to the minimum failing product sample.

**Comment by phortx** [comment]
I think, I found the reason. In a rather long file there was something like this:

```slim
span.icon.ion-trash-o"
```

Also I found something like this:

```slim
span.icon.ion-trash-o>
```

The chars are an artifact from converting HTML to Emblem. I've searched for that pattern in all files and removed the chars. After that, everything works.

However it's hard to debug that. Better error messages would be nice! :)
